src/textual/widgets/_digits.py

```diff
@@ -0,0 +1,94 @@
+from __future__ import annotations
+
+from typing import cast
+
+from rich.align import Align, AlignMethod
+from rich.console import RenderableType
+
+from ..geometry import Size
+from ..renderables.digits import Digits as DigitsRenderable
+from ..widget import Widget
+
+
+class Digits(Widget):
+    """A widget to display numerical values using a 3x3 grid of unicode characters.
+
+    Args:
+        value: Value to display in widget.
+        name: The name of the widget.
+        id: The ID of the widget in the DOM.
+        classes: The CSS classes of the widget.
+        disabled: Whether the widget is disabled or not.
+
+    """
+
+    DEFAULT_CSS = """
+    Digits {
+        width: 1fr;
+        height: auto;
+        text-align: left;
+        text-style: bold;
+        box-sizing: border-box;
+    }
+    """
+
+    def __init__(
+        self,
+        value: str = "",
+        *,
+        name: str | None = None,
+        id: str | None = None,
+        classes: str | None = None,
+        disabled: bool = False,
+    ) -> None:
+        if not isinstance(value, str):
+            raise TypeError("value must be a str")
+        super().__init__(name=name, id=id, classes=classes, disabled=disabled)
+        self.value = value
+
+    def update(self, value: str) -> None:
+        """Update the Digits with a new value.
+
+        Args:
+            value: New value to display.
+        """
+        if not isinstance(value, str):
+            raise TypeError("value must be a str")
```

Comment:
Needs a `Raises` in the docstring.

Also, as mentioned for `__init__`, I can see some utility in allow `str | float` for this.